src/extensions/FullscreenExtension/FullscreenExtension.ts

```diff
@@ -30,12 +30,26 @@ export class FullscreenExtension extends Module {
         } as IFullscreenEventData);
       });
     }
+    else if(instance.env.isIOS) {
+      this.on(Events.READY, this.onReady.bind(this));
+    }
+    
+    
+  }
+  
+  public onReady() {
+    if(this.instance.player.mediaElement.webkitEnterFullscreen) {
```

Comment:
I think it's best to abstract the different fullscreen possibilities (eg: native with fullscreen, native on the video element on iOS and a fake fullscreen with a fixed container). Based on a simple rule set, an implementation can be selected.

Using webkitEnterFullscreen would disable certain features on google-ima and I'm not sure that's desired. If so, we can always make this an opt-in in the config (eg: allowIOSNativeFullscreen).

Would you mind splitting this logic in a separate PR?